### PR TITLE
[jvm-packages] fix executor crashing issue when transforming on xgboost4j-spark-gpu

### PR DESCRIPTION
**Pull request by wbo4958**
the API XGBoosterSetParam is not thread-safe. During the phase of transforming,
XGBoost runs several transforming tasks at a time, and each of them will set
the "gpu_id" and "predictor" parameters, so if several tasks (multi-threads)
all XGBoosterSetParam simultaneously, it may cause the memory to be corrupted
and cause SIGSEGV.

This PR first gets the booster from broadcast and set to the correct gpu_id
and predictor, and then all transforming tasks will use the same booster to
do the transforming.

To fix https://github.com/dmlc/xgboost/issues/8016

### CODE REVIEW

**Comment by wbo4958** [comment]
@trivialfis could you help to review it?

File: jvm-packages/xgboost4j-spark-gpu/src/main/scala/ml/dmlc/xgboost4j/scala/rapids/spark/GpuPreXGBoost.scala [hunk 1]
```diff
@@ -218,7 +223,7 @@ object GpuPreXGBoost extends PreXGBoostProvider {
 
         // predict and turn to Row
         val predictFunc =
-          (broadcastBooster: Broadcast[Booster], dm: DMatrix, originalRowItr: Iterator[Row]) => {
+          (broadcastBooster: Booster, dm: DMatrix, originalRowItr: Iterator[Row]) => {
```
[the remark this answers is not in the document]
Comment (wbo4958) [reply jvm-packages/xgboost4j-spark-gpu/src/main/scala/ml/dmlc/xgboost4j/scala/rapids/spark/GpuPreXGBoost.scala]: yeah, good catch. fixed

**Review by trivialfis — APPROVED** [review]

**Comment by haohenggang** [comment]
Is there any xgboost4j.jar available for Android phones that we would like to train online on?

**Comment by wbo4958** [comment]
Seems xgboost doesn't support ARM platform? @trivialfis

**Comment by trivialfis** [comment]
We do run some simple tests on ARM. The full test suite is not run due to CI time constraints, but I don't expect any difficulty.

**Pull request merged**